gc_derive/src/lib.rs

```diff
@@ -51,23 +56,41 @@ fn derive_trace(mut s: Structure<'_>) -> proc_macro2::TokenStream {
         },
     );
 
-    // We also implement drop to prevent unsafe drop implementations on this
-    // type and encourage people to use Finalize. This implementation will
-    // call `Finalize::finalize` if it is safe to do so.
-    let drop_impl = s.unbound_impl(
-        quote!(::std::ops::Drop),
+    if !is_trivially_drop {
+        // We also implement drop to prevent unsafe drop implementations on this
+        // type and encourage people to use Finalize. This implementation will
+        // call `Finalize::finalize` if it is safe to do so.
+        let drop_impl = s.unbound_impl(
+            quote!(::std::ops::Drop),
+            quote! {
+                fn drop(&mut self) {
+                    if ::gc::finalizer_safe() {
+                        ::gc::Finalize::finalize(self);
+                    }
+                }
+            },
+        );
+
         quote! {
-            fn drop(&mut self) {
-                if ::gc::finalizer_safe() {
-                    ::gc::Finalize::finalize(self);
+            #trace_impl
+            #drop_impl
+        }
+    } else {
+        s.bind_with(|_| BindStyle::Move);
+        let trivially_drop_body = s.each(|bi| quote!(drop(#bi)));
```

Comment:
This works, but simpler is
```suggestion
        let trivially_drop_body = s.each(|_| quote! {});
```